src/components/product-filters/product-filters.tsx

```diff
@@ -53,13 +53,18 @@ export const ProductFilters = ({
                     content: (
                         <RangeSlider
                             className="rangeSlider"
+                            step="any"
                             startValue={filters.minPrice ?? lowestPrice}
                             endValue={filters.maxPrice ?? highestPrice}
                             onStartValueChange={(value) => {
-                                handleFiltersChange({ minPrice: value });
+                                handleFiltersChange({
+                                    minPrice: value === lowestPrice ? value : Math.round(value),
```

Comment:
I think it should be `Math.floor` for the lower bound and `Math.ceil` for the upper bound to ensure we're not excluding results in the filter's range.